frontend/src/store.js

```diff
@@ -0,0 +1,34 @@
+import Vue from 'vue'
+import Vuex from 'vuex'
+
+Vue.use(Vuex)
+
+// Create enum style variables (to reduce typo errors)
+const MUTATIONS = Object.freeze({
+  SET_STUDENTS: 'SET_STUDENTS'
+})
+
+export default new Vuex.Store({
+  state: {
+    students: []
+  },
+  mutations: {
+    [MUTATIONS.SET_STUDENTS](state, students) {
+      state.students = students
+    }
+  },
+  actions: {
+    async getAllStudents({ commit }) {
+      await fetch('/api/students')
+        .then(response => {
+          return response.json();
+        }).then(data => {
+          commit(MUTATIONS.SET_STUDENTS, data)
+        })
```

Comment:
The `async/await` syntax and `.then` syntax both handle asynchronous actions (like fetch). However, it's best not to mix the two syntaxes. Consider that these 2 are equivalent:
```js
async getAllStudents() {
  const response = await fetch("/api/students")
  const data = await response.json()
  commit(MUTATIONS.SET_STUDENTS, data)
}
```
and
```js
getAllStudents() {
  return fetch("/api/students")
    .then(response => response.json())
    .then(data => commit(MUTATIONS.SET_STUDENTS, data))
}
```
Both `fetch` function and the `.json()` method are asynchronous, that is why you must either `await` them, or use the `.then` before moving onto the "next thing" you want to do.

Reply:
Oh i was mixing them up like a messy salad! Thanks for your examples. Cleared up my understanding. Have cleaned up the code.